frame/pov-limit/src/weights.rs

```diff
@@ -0,0 +1,92 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Autogenerated weights for pallet_pov_limit
+//!
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
+//! DATE: 2022-12-21, STEPS: `50`, REPEAT: 20, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! HOSTNAME: `Sergejs-MacBook-Air.local`, CPU: `<UNKNOWN>`
+//! EXECUTION: Some(Wasm), WASM-EXECUTION: Compiled, CHAIN: Some("dev"), DB CACHE: 1024
+
+// Executed Command:
+// ./target/debug/substrate
+// benchmark
+// pallet
+// --chain=dev
+// --steps=50
+// --repeat=20
+// --pallet=pallet_pov_limit
+// --execution=wasm
+// --wasm-execution=compiled
+// --heap-pages=4096
+// --output=./frame/pov-limit/src/weights.rs
+// --header=./HEADER-APACHE2
+// --template=./.maintain/frame-weight-template.hbs
+// --extrinsic=*
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::{traits::Get, weights::{Weight, constants::RocksDbWeight}};
+use sp_std::marker::PhantomData;
+
+/// Weight functions needed for pallet_pov_limit.
+pub trait WeightInfo {
+	fn hash_value() -> Weight;
+	fn read() -> Weight;
+	fn on_idle() -> Weight;
+}
+
+/// Weights for pallet_pov_limit using the Substrate node and recommended hardware.
+pub struct SubstrateWeight<T>(PhantomData<T>);
+impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	fn hash_value() -> Weight {
+		// Minimum execution time: 13_000 nanoseconds.
+		Weight::from_ref_time(14_000_000)
+	}
+	fn read() -> Weight {
+		// Minimum execution time: 3_000 nanoseconds.
+		Weight::from_ref_time(3_000_000)
+	}
+	// Storage: PovLimit Storage (r:1 w:0)
+	// Storage: PovLimit Compute (r:1 w:0)
+	fn on_idle() -> Weight {
+		// Minimum execution time: 260_000 nanoseconds.
+		Weight::from_ref_time(262_000_000)
+			.saturating_add(T::DbWeight::get().reads(2))
```

Comment:
@ggwpez It seems like the `on_idle` takes a bit more time than it is supposed to(it should take around `200µs` but it seems like it is taking about `262µs`). Do you think it is possible that a few additions and if statements(for which the execution time is not tracked) could make this difference? Or am I measuring the weight inside the function wrongly?

Reply:
I added an `on_idle_weight_is_close_enough_works` test which should ensure that we consume the correct amount.  
Now we also need a test that checks that the benchmarked result is close enought.  
So the result of `T::WeightInfo::on_idle` needs to be checked as well. Only then can we start to try out different approaches to fix it, otherwise its difficult to measure the effect.